shared-bindings/os/getenv.c

```diff
@@ -0,0 +1 @@
+// this file needs to exist but does not have any content
```

Comment:
Why does this need to exist? Usually files here are either __init__ or a class. This is weird to have for a function.

Reply:
It's split out because this implementation (but not the rest of os) needs to be available in the unix port.

but when I added the pattern that would pick up `shared-module/os/getenv.c` during build (in `circuitpy_defns.mk` `SRC_SHARED_MODULE_ALL` variable) builds failed because they refered to this filename. Do I need to list it differently?

Reply:
I think you solved the pattern problem. The function could be moved back to `__init__.c` if the file still poses an issue.

Reply:
You don't need this file if you use SRC_SHARED_MODULE_INTERNAL: https://github.com/adafruit/circuitpython/blob/main/py/circuitpy_defns.mk#L720

I'd put the common_hal_ functions into `__init__.c` in shared-module and the parsing stuff can stay in the "internal file" getenv.c.

Reply:
I will do the 'fix' for needing the nearly-empty file.

Most of the code has to stay in getenv.c so that it can be linked into the unix coverage build for testing.